drivers/flash/spi_nor.c

```diff
@@ -554,15 +602,72 @@ static int exit_dpd(const struct device *const dev)
 	return ret;
 }
 
-/* Everything necessary to acquire owning access to the device. */
-static void acquire_device(const struct device *dev)
+#ifdef CONFIG_SPI_NOR_SUSPEND_RESUME
+static int spi_nor_suspend_resume(const struct device *dev)
+{
+	struct spi_nor_data *const driver_data = dev->data;
+	uint8_t opcode;
+	int ret;
+
+	if (driver_data->suspended) {
+		opcode = driver_data->operation & SPI_NOR_ERASE ? driver_data->resm_instr
+								: driver_data->presm_instr;
+	} else {
+		opcode = driver_data->operation & SPI_NOR_ERASE ? driver_data->susp_instr
+								: driver_data->psusp_instr;
+	}
+
+	ret = spi_nor_cmd_write(dev, opcode);
+	if (ret != 0) {
+		return ret;
+	}
+
+	while (true) {
+		uint8_t reg;
+
+		ret = spi_nor_cmd_read(dev, SPI_NOR_CMD_RDSR, &reg, sizeof(reg));
+		/* Exit on error or no longer WIP */
+		if (ret || !(reg & SPI_NOR_WIP_BIT)) {
+			break;
+		}
+
+		k_sleep(WAIT_READY_SUSPEND_RESUME);
```

Comment:
I think that is also returned by the BFP table, but I am not sure whether we want to read this and process, probably would be optimal but would increase size of PR.

Reply:
There should be some timeout here probably, so that we can bail out if we keep on getting WIP == 1 for too long.